backend/ethereum/subscription/eventsub.go

```diff
@@ -199,25 +205,25 @@ func (s *EventSub) readFuture(ctx context.Context, sink chan<- *Event) error {
 			case <-ctx.Done():
 				return ctx.Err()
 			case sink <- event:
-			case <-s.closed:
+			case <-s.closer.Closed():
 				return nil
 			}
 		case err := <-s.watchSub.Err():
 			err = cherrors.CheckIsChainNotReachableError(err)
 			return err
 		case <-ctx.Done():
 			return ctx.Err()
-		case <-s.closed:
+		case <-s.closer.Closed():
 			return nil
 		}
 	}
 }
 
 // Close closes the sub and frees associated resources.
-// Should be called exactly once and panics otherwise.
-// Must not be called if the construction function returned with an error.
+// Can be called more than once, but should be called at least once.
```

Comment:
I think the second part is clear.
```suggestion
// Can be called more than once.
```